src/v2/Carousel/Carousel.sol

```diff
@@ -0,0 +1,521 @@
+        ) revert OwnerDidNotAuthorize(msg.sender, _receiver);
+
+        // swich the last item in the queue with the item to be removed
+        uint256 index = getRolloverIndex(_receiver);
+        if (index == rolloverQueue.length - 1) {
+            rolloverQueue.pop();
+            delete ownerToRollOverQueueIndex[_receiver];
+        } else {
+            // overwrite the item to be removed with the last item in the queue
+            rolloverQueue[index] = rolloverQueue[rolloverQueue.length - 1];
+            // remove the last item in the queue
+            rolloverQueue.pop();
+            // update the index of prev last user
+            ownerToRollOverQueueIndex[rolloverQueue[index].receiver] = index;
+            // remove receiver from index mapping
+            delete ownerToRollOverQueueIndex[_receiver];
+        }
+    }
+
+    function mintDepositInQueue(uint256 _epochId, uint256 _operations)
+        external
+        epochIdExists(_epochId)
+        epochHasNotStarted(_epochId)
+        nonReentrant
+    {
+        // make sure there is already a new epoch set
+        // epoch has not started
+        QueueItem[] memory queue = depositQueue;
+        uint256 length = rolloverQueue.length;
+
+        // revert if queue is empty or operations are more than queue length
+        if (length == 0 || _operations > length - 1) revert OverflowQueue();
+
+        // queue is executed from the tail to the head
+        // get last index of queue
+        uint256 i = length - 1;
+        while (i > (length - 1) - _operations) {
+            _mint(
+                queue[i].receiver,
+                queue[i].epochId,
+                queue[i].assets - relayerFee,
+                EMPTY
+            );
+            depositQueue.pop();
+            unchecked {
+                i--;
+            }
+        }
+
+        asset.safeTransfer(feeTreasury, _operations * relayerFee);
+    }
+
+    function mintRollovers(uint256 _epochId, uint256 _operations)
+        external
+        epochIdExists(_epochId)
+        epochHasNotStarted(_epochId)
+        nonReentrant
+    {
+        // make sure there is already a new epoch set
+        // epoch has not started
+        // prev epoch is resolved
+        if (
+            epochResolved[epochs[epochs.length - 2]] &&
+            epochs[epochs.length - 1] == _epochId
+        ) {
+            QueueItem[] memory queue = rolloverQueue;
+            uint256 length = rolloverQueue.length;
+            // revert if queue is empty or operations are more than queue length
+            if (length == 0 || _operations > length - 1) revert OverflowQueue();
+            // account for how many operations have been done
+            uint256 index = rolloverAccounting[_epochId];
+            while (index < _operations) {
+                // only roll over if user won last epoch
+                if (
+                    previewWithdraw(queue[index].epochId, queue[index].assets) >
```

Comment:
might be cleaner to abstract this check to, `hasUserWonEpoch(id)` (would be useful external read too